cas-server-core/src/main/java/org/jasig/cas/authentication/principal/WebApplicationServiceFactory.java

```diff
@@ -55,9 +55,12 @@ public WebApplicationService createService(final HttpServletRequest request) {
         final String id = cleanupUrl(serviceToUse);
         final String artifactId = request.getParameter(CasProtocolConstants.PARAMETER_TICKET);
 
-        return new SimpleWebApplicationServiceImpl(id, serviceToUse,
-                artifactId, "POST".equalsIgnoreCase(method) ? Response.ResponseType.POST
-                : Response.ResponseType.REDIRECT);
+        final  Response.ResponseType type = "POST".equalsIgnoreCase(method) ? Response.ResponseType.POST
+                                                        : Response.ResponseType.REDIRECT;
+
+        final WebApplicationService webApplicationService = new SimpleWebApplicationServiceImpl(id, serviceToUse,
+                artifactId, new WebApplicationServiceResponseBuilder(type));
```

Comment:
I would say, yes because each serve may produce a response that is different. For example, openid responses are very different from SAML responses, and to produce one, they each need different dependencies. So we are mapping services to response for each service. This also has the advantage that modifying on response builder wouldnt affect another, which wouldnt be the case if we just have one generic builder.